src/stan/mcmc/hmc/hamiltonians/base_hamiltonian.hpp

```diff
@@ -36,6 +36,8 @@ namespace stan {
         return T(z) + V(z);
       }
 
+      virtual double dG_dt(Point& z) = 0;
```

Comment:
They actually can't be made const as the Point objects also hold auxiliary state to avoid recomputation, such as the gradients and, for Riemannian HMC, metric and other intermediate variables.  Many of these classes will modify those internal states.